jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java

```diff
@@ -779,6 +790,19 @@ public Response payAllInvoices(@PathParam("accountId") final UUID accountId,
                                              null :
                                              (inputPaymentMethodId != null ? inputPaymentMethodId : account.getPaymentMethodId());
                 createPurchaseForInvoice(account, invoice.getId(), amountToPay, paymentMethodId, externalPayment, null, null, pluginProperties, callContext);
+
+                if (filterMinDate == null || filterMinDate.isAfter(invoice.getTargetDate())) {
```

Comment:
Is it really the target date we want to look at here? Or the invoice date?

Reply:
Should be the targetDate -- see https://github.com/killbill/killbill/blob/master/invoice/src/main/resources/org/killbill/billing/invoice/dao/InvoiceSqlDao.sql.stg#L90